version: '3.7'

services:
  updater:
    build: ./service_updater
    command: bash -c 'python tasks.py'
    environment:
      - DATABASE_URL=postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_NAME}
    restart: always
    volumes:
      - /home/user_bot/disk-z:/files
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro

  service_customers:
    build: ./service_customers/
    container_name: service_customers
    ports:
      - 9000:9000
    volumes:
      - /home/user_bot/databases/:/databases
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro

  service_control:
    build: ./service_control/
    container_name: service_control
    ports:
      - 8500:8500
    volumes:
      - /home/user_bot/disk-z:/files
      - /home/user_bot/databases/:/databases
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro

  service_bot:
    build: ./service_bot
    container_name: service_bot
    restart: always
    network_mode: host
    volumes:
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro

